kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/consumer/streamable/KafkaTrackingToken.java

```diff
@@ -111,95 +101,83 @@ public static boolean isEmpty(KafkaTrackingToken token) {
-
+        //noinspection ConstantConditions - Verified cast through `Assert.isTrue` operation
         return new KafkaTrackingToken(bounds((KafkaTrackingToken) other, Math::min));
     }
 
-    @SuppressWarnings("ConstantConditions") // Verified TrackingToken type through `Assert.isTrue` operation
     @Override
     public TrackingToken upperBound(TrackingToken other) {
         isTrue(other instanceof KafkaTrackingToken, () -> "Incompatible token type provided.");
-
+        //noinspection ConstantConditions - Verified cast through `Assert.isTrue` operation
         return new KafkaTrackingToken(bounds((KafkaTrackingToken) other, Math::max));
     }
 
-    private Map<Integer, Long> bounds(KafkaTrackingToken token, BiFunction<Long, Long, Long> boundsFunction) {
-        Map<Integer, Long> intersection = new HashMap<>(this.partitionPositions);
-        token.partitionPositions().forEach(intersection::putIfAbsent);
+    private Map<TopicPartition, Long> bounds(KafkaTrackingToken other, BiFunction<Long, Long, Long> boundsFunction) {
+        Map<TopicPartition, Long> intersection = new HashMap<>(tokenPartitionPositions());
+        other.tokenPartitionPositions().forEach(intersection::putIfAbsent);
 
         intersection.keySet()
-                .forEach(partitionNumber -> intersection.put(partitionNumber, boundsFunction.apply(
-                        this.partitionPositions().getOrDefault(partitionNumber, 0L),
-                        token.partitionPositions().getOrDefault(partitionNumber, 0L))
-
-                ));
+                    .forEach(topicPartition -> intersection.put(topicPartition, boundsFunction.apply(
+                            this.tokenPartitionPositions().getOrDefault(topicPartition, 0L),
+                            other.tokenPartitionPositions().getOrDefault(topicPartition, 0L))
+                    ));
         return intersection;
     }
 
-    @SuppressWarnings("ConstantConditions") // Verified TrackingToken type through `Assert.isTrue` operation
     @Override
     public boolean covers(TrackingToken other) {
         isTrue(other instanceof KafkaTrackingToken, () -> "Incompatible token type provided.");
+        //noinspection ConstantConditions - Verified cast through `Assert.isTrue` operation
         KafkaTrackingToken otherToken = (KafkaTrackingToken) other;
 
-        return otherToken.partitionPositions
-                .entrySet().stream()
-                .allMatch(position -> position.getValue() <= this.partitionPositions
-                        .getOrDefault(position.getKey(), -1L));
+        return otherToken.tokenPartitionPositions()
+                         .entrySet().stream()
+                         .allMatch(offsetsEntry -> match(offsetsEntry.getKey(), offsetsEntry.getValue()));
+    }
+
+    private boolean match(TopicPartition otherTopicPartition, long otherOffset) {
+        return otherOffset <= this.tokenPartitionPositions().getOrDefault(otherTopicPartition, -1L);
```

Comment:
Would you consider moving this method to `TopicPartition`?

Reply:
The `TopicPartition` is a Kafka class, so I couldn't! It would however be a nice-to-have.

Reply:
Oh :D